### PR TITLE
Add effort section to DG

### LINKED ISSUE #356: Update effort section in DG

### PR DESCRIPTION
**Pull request by ngjianyi**
Closes #356

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -895,3 +895,32 @@ testers are expected to do more *exploratory* testing.
...
        Expected: The app closes.
+
+## **Appendix: Effort**
+
+### Difficulty Level
+VendorVault is significantly more complex than Address Book 3 (AB3). While AB3 deals with a single entity (Person), 
+VendorVault deals with two entities (Supplier and Delivery), thus more effort is required to manage and integrate the two entities seamlessly. 
+Apart from introducing new commands, the UI was also improved to support displaying both entities in a user-friendly manner.
+
+### Challenges Faced
```
Comment (CJianzhi) [inline docs/DeveloperGuide.md:906]: Can we write about having a separate list for sorting as a challenge?

**Review by CJianzhi — APPROVED** [review]
LGTM! For challenges maybe we can write about the difficulties in implementing specific commands?

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (LinkesV) [reply docs/DeveloperGuide.md:906]: I think it is ok

**Review by LinkesV — APPROVED** [review]
LGTM

**Pull request merged**